portal-ui/src/screens/Console/Tenants/TenantDetails/TenantMonitoring.tsx

```diff
@@ -0,0 +1,324 @@
+  });
+
+const TenantMonitoring = ({
+  classes,
+  match,
+  tenant,
+  loadingTenant,
+}: ITenantMonitoring) => {
+  const [prometheusMonitoringEnabled, setPrometheusMonitoringEnabled] =
+    useState<boolean>(false);
+  const [edit, setEdit] = useState<boolean>(false);
+  const [monitoringInfo, setMonitoringInfo] =
+    useState<ITenantMonitoringStruct>();
+  const [confirmOpen, setConfirmOpen] = useState<boolean>(false);
+  const [refreshMonitoringInfo, setRefreshMonitoringInfo] =
+    useState<boolean>(true);
+
+  const tenantName = match.params["tenantName"];
+  const tenantNamespace = match.params["tenantNamespace"];
+
+  const onCloseEditAndRefresh = () => {
+    setEdit(false);
+    setRefreshMonitoringInfo(true);
+  };
+
+  useEffect(() => {
+    if (refreshMonitoringInfo) {
+      api
+        .invoke(
+          "GET",
+          `/api/v1/namespaces/${tenantNamespace}/tenants/${tenantName}/monitoring`
+        )
+        .then((res: ITenantMonitoringStruct) => {
+          setPrometheusMonitoringEnabled(res.prometheusEnabled);
+          setMonitoringInfo(res);
+          setRefreshMonitoringInfo(false);
+        })
+        .catch((err: ErrorResponseHandler) => {
+          setErrorSnackMessage(err);
+          setRefreshMonitoringInfo(false);
+        });
+    }
+  }, [refreshMonitoringInfo]);
+
+  const togglePrometheus = () => {
+    const configInfo = {
+      prometheusEnabled: prometheusMonitoringEnabled,
+      toggle: true,
+    };
+    api
+      .invoke(
+        "PUT",
+        `/api/v1/namespaces/${tenantNamespace}/tenants/${tenantName}/monitoring`,
+        configInfo
+      )
+      .then(() => {
+        setPrometheusMonitoringEnabled(!prometheusMonitoringEnabled);
+        setRefreshMonitoringInfo(true);
+        setConfirmOpen(false);
+        setRefreshMonitoringInfo(true);
+      })
+      .catch((err: ErrorResponseHandler) => {
+        setErrorSnackMessage(err);
+      });
+  };
+
+  return (
+    <Fragment>
+      {edit && tenant !== null && prometheusMonitoringEnabled && (
+        <EditTenantMonitoringModal
+          classes={classes}
+          open={edit}
+          onClose={onCloseEditAndRefresh}
+          tenant={tenant}
+          image={monitoringInfo?.image || ""}
+          sidecarImage={monitoringInfo?.sidecarImage || ""}
+          initImage={monitoringInfo?.initImage || ""}
+          diskCapacityGB={
+            monitoringInfo?.diskCapacityGB
+              ? parseInt(monitoringInfo?.diskCapacityGB)
+              : 5
+          }
+          labels={monitoringInfo?.labels || []}
+          annotations={monitoringInfo?.annotations || []}
+          nodeSelector={monitoringInfo?.nodeSelector || []}
+          serviceAccountName={monitoringInfo?.serviceAccountName || ""}
+          tenantName={tenantName}
+          tenantNamespace={tenantNamespace}
+          storageClassName={monitoringInfo?.storageClassName || ""}
+        />
+      )}
+      {confirmOpen && (
```

Comment:
based on the new import:
```suggestion
     {confirmOpen && (
        <ConfirmDialog
          isOpen={confirmOpen}
          title={
            prometheusMonitoringEnabled
              ? "Disable Prometheus monitoring?"
              : "Enable Prometheus monitoring?"
          }
          confirmText={prometheusMonitoringEnabled ? "Disable" : "Enable"}
          cancelText="Cancel"
          onClose={() => setConfirmOpen(false)}
          onConfirm={togglePrometheus}
          confirmationContent={
            <DialogContentText>
              {prometheusMonitoringEnabled
                ? "Disabling monitoring will erase any custom values you have used to configure Prometheus monitoring"
                : "Prometheus monitoring will be enabled with default values"}
            </DialogContentText>
          }
        />
      )}
```